src/wp-includes/class-wp-query.php

```diff
@@ -1862,11 +1862,7 @@ public function get_posts() {
 		}
 
 		if ( ! isset( $q['cache_results'] ) ) {
-			if ( wp_using_ext_object_cache() ) {
-				$q['cache_results'] = false;
-			} else {
-				$q['cache_results'] = true;
-			}
+			$q['cache_results'] = true;
```

Comment:
The old code here is very odd, I'm going to track down the history if I can. Mainly curious 🐱 -- i think the change makes sense.

Reply:
Added in 14d8731a4998ad61c064b6de8506f2aa797d6092 (r14665 on trac). I suspect it could have been updated years ago following 021c55798ecf07517c7450a119ec6e2b52e7c352 (r19918 on trac) but never was. 

Cool story, Wilson. :D

Reply:
I think it is safe to change this. It is very weird as is